src/CSVReader/EventSource.cs

```diff
@@ -90,16 +95,22 @@ public abstract class EventSource
 
         /// <summary>
         /// Utility function that takes a specification for columns (which can include *) and the raw list of
-        /// all possible columns and returns the list that match the specification. 
+        /// all possible columns and returns the list that match the specification. This function also extracts the event filter query and creates a FilterQueryExpressionTree.
         /// </summary>
-        public static List<string> ParseColumns(string columnSpec, ICollection<string> columnNames)
+        public static List<string> ParseColumns(string columnSpec, ICollection<string> columnNames, out FilterQueryExpressionTree filterQueryExpressionTree)
```

Comment:
This looks like a breaking change for any extension that might use the events view to show non-standard data.  Do you think that it is possible to remove the new parameter and depend on the `FilterQueryExpressionTree` property?  From my review so far, I'm not sure how to think about the property vs. this out parameter.